example/advection/advection_package.cpp

```diff
@@ -140,42 +143,56 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   pkg->AddParam<>("sin_a3", sin_a3);
 
   // number of variable in variable vector
+  const auto vec_size = pin->GetOrAddInteger("Advection", "vec_size", 1);
   const auto num_vars = pin->GetOrAddInteger("Advection", "num_vars", 1);
+  pkg->AddParam<>("vec_size", vec_size);
+  pkg->AddParam<>("num_vars", num_vars);
 
-  std::string field_name = "advected";
   // Give a custom labels to advected in the data output
-  std::vector<std::string> advected_labels;
-  advected_labels.reserve(num_vars);
-  for (int i = 0; i < num_vars; i++) {
-    advected_labels.push_back("Advected_" + std::to_string(i));
+  std::string field_name_base = "advected";
+  std::string field_name;
+  Metadata m;
+  for (int var = 0; var < num_vars; ++var) {
+    std::vector<std::string> advected_labels;
+    advected_labels.reserve(vec_size);
+    for (int j = 0; j < vec_size; ++j) {
+      advected_labels.push_back("Advected_" + std::to_string(var) + " _" +
+                                std::to_string(j));
```

Comment:
We usually use the label format "variable_x" to denote sparse variables with index x. Of course our infrastructure will have to be robust against a dense variables with such names, but maybe to avoid confusion we should use "variable-x" or "variableX" here?

Reply:
Well taken. I'll add an issue to address this when I improve the automated tests